nservicebus/extending/features.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Feature-based extensibility system
+summary: Explains the concept of feature
+tags:
+ - feature
+ - extensibility
+ - extending
+---
+
+NServiceBus code base consists of two parts.
+
+### Bootstrapping code
+
+The bootstrapping code is responsible for taking the configuration values from the user (via the `BusConfiguration`) API and constructing and starting a bus. The bus is constructed by activating features.
+
+### Message processing code
+
+All the actual logic of the bus is implemented in form of features. Features take part in configuring the underlying dependency injection container with all the services required by the bus to be operational.
+
+## Definition
+
+A feature is defined in a class inheriting from `Feature`. This is a minimal feature that does nothing
+
+<!-- import MinimalFeature -->
+
+
+## Dependencies
+
+Features are so powerful because they can depend on each other to form more complex behaviours.
+
+<!-- import DependentFeature -->
+
+A feature might use either strongly or loosely typed API when declaring dependencies (the latter can be useful if a dependency is declared in an external assembly).
+
+NOTICE: The feature name is derived from the name of the type but the suffix `Feature`, if present in the type name, is removed from the name of the feature.
```

Comment:
"notice". do u mean "note:". also there is no context as to why the "feature name" is important